src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,298 @@
+/** @jsxImportSource @emotion/react */
+import { css } from "@emotion/react";
+import {
+  DragEvent,
+  DragEventHandler,
+  useContext,
+  useRef,
+  useState
+} from "react";
+import { IoIosArrowDown, IoIosArrowForward } from "react-icons/io";
+import { DragInfo } from "./Tree";
+import TreeContext from "./TreeContext";
+import TreeNode from "./treeNode";
+
+type DroppableState = {
+  prev: boolean;
+  this: boolean;
+  label: boolean;
+  next: boolean;
+};
+
+const DROP_DISABLED = {
+  prev: false,
+  this: false,
+  label: false,
+  next: false
+} as const satisfies DroppableState;
+
+type Props = {
+  node: TreeNode;
+  dragInfo: DragInfo;
+  onDragStart: (nodeId: string, previousNodeId: string | undefined) => void;
+  previousNodeId: string | undefined;
+  ancestorNodeIds?: string[];
+  isFirstNode?: boolean;
+};
+
+const TreeItem = ({
+  node,
+  dragInfo,
+  onDragStart,
+  previousNodeId,
+  ancestorNodeIds = [],
+  isFirstNode = false
+}: Props) => {
+
+  const isRoot = ancestorNodeIds.length === 0;
+
+  const context = useContext(TreeContext);
+
+  const isSelected = context.selectedNodeId === node.id;
+
+  const handleSelect = () => {
+    context.onSelect && context.onSelect(node.id);
```

Comment:
tipsですが、ここはこのように書けます。

```suggestion
    context?.onSelect(node.id);
```

Reply:
ありがとうございます。
忘れていて、力業で実装していました。
修正します。